### DISCUSSION TITLE
how to autoscroll

### DISCUSSION THREAD

**Discussion by highspeeder** [AvaloniaUI/Avalonia#9236] [Q&A] [answered]
Whenever the bound Log property changes
I want to bring down the scroll of ScrollViewer to the bottom.

I tried to implement it using Avalonia.Xaml.Behaviors but failed.

Any ideas?

```
<ScrollViewer x:Name="LogSV" VerticalScrollBarVisibility="Auto" HorizontalScrollBarVisibility="Auto">
    <TextBox x:Name="LogText" Text="{Binding Log}" IsReadOnly="True" Focusable="False"/>
</ScrollViewer>
```

If the textbox has a textChanged event, it seems to be easy to achieve.

**Comment by timunie** [comment, marked as answer]
Ah for logs I prefer not to use TextBox at all because it's really slow with long text. As an alternative you can use a ListBox or AvaloniaEdit which supports more text. Scrolling is also possible.
